### PR TITLE
Schema update/Improved validation & error handling

### PR DESCRIPTION
**Pull request by kyle1morel**
# Description

Updates the database schema
- Adds `initiative` and `activity` tables
- Move to underscored naming instead of camel case

Updates to error handling and validation throughout.

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by wilwong89 — APPROVED** [review]

**Review by jatindersingh93 — APPROVED** [review]

File: frontend/src/components/submission/SubmissionForm.vue [hunk 1]
```diff
@@ -88,11 +90,22 @@ const formSchema = object({
     .label('Submission Types'),
   user: mixed()
     .when('intakeStatus', {
-      is: (val: string) => val !== 'SUBMITTED',
-      then: (schema) => schema.required(),
-      otherwise: (schema) => schema.notRequired()
+      is: (val: string) => val === INTAKE_STATUS_LIST.SUBMITTED,
+      then: (schema) =>
+        schema
+          .test('expect-user-or-empty', 'Assigned to must be empty or a selected user', (obj) => {
```
Comment (TimCsaky) [inline frontend/src/components/submission/SubmissionForm.vue:96]: you could rename `obj` to `input` to make it more readable:
`.test('expect-user-or-empty', 'Assigned to must be empty or a selected user', (input) => {`

File: frontend/src/types/Statistics.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  queue_1: number;
+  queue_2: number;
+  queue_3: number;
+  queue_4: number;
+  queue_5: number;
+  guidance: number;
+  status_request: number;
+  inquiry: number;
+  emergency_assist: number;
+  inapplicable: number;
```
Comment (TimCsaky) [inline frontend/src/types/Statistics.ts:24]: i'm guessing you intentionally made just these types names with underscores

**Review by TimCsaky — APPROVED** [review]
wow.. a lot of name changes 😅 good work!

File: app/src/db/migrations/20231212000000_init.ts [hunk 3]
```diff
@@ -29,167 +29,197 @@ export async function up(knex: Knex): Promise<void> {
...
-          table.text('locationPIDs');
-          table.text('companyNameRegistered');
-          table.text('contactName');
-          table.text('contactPhoneNumber');
-          table.text('contactEmail');
-          table.text('projectName');
-          table.text('singleFamilyUnits');
-          table.text('streetAddress');
+          table.uuid('submission_id').primary();
+          table.text('activity_id').notNullable();
```
[the remark this answers is not in the document]
Comment (kyle1morel) [reply app/src/db/migrations/20231212000000_init.ts:73]: Wow. Yeah, I completely missed creating the activity entry. And yes, I'll make it a FK.

File: frontend/src/types/Statistics.ts (hunk 2 above)
Comment (kyle1morel) [reply frontend/src/types/Statistics.ts:24]: Yeah these should be converted to camel case before being sent to the client..

**Comment by kyle1morel** [comment]
Closing. This line of work will be part of a different PR.

**Pull request closed without merging**